### PR TITLE
Added media scrolling

### PR DESCRIPTION
**Pull request by Revadike**
Resolves https://github.com/tfedor/AugmentedSteam/issues/424

### CODE REVIEW

**Comment by MxtOUT** [comment]
I don't see it working or am I looking at the wrong place?
Trying it on app pages

**Comment by Revadike** [comment]
> I don't see it working or am I looking at the wrong place?
> Trying it on app pages

Go to a store app page, hover with your mouse on the media thumbnails bar and start scrolling. The reason why it's only activated on that bar, is to prevent annoyance when you actually want to scroll to the bottom of the page.

**Comment by MxtOUT** [comment]
Got it, thanks!

**Comment by Revadike** [comment]
> The formatting is somehow messed up

Where? Can you be a bit more specific?

File: js/content/common.js [hunk 1]
```diff
@@ -2282,5 +2282,34 @@ class MediaPage {
...
+        let strip = document.querySelector("#highlight_strip");
+        if (!strip) { return; }
+
+		let lastScroll = Date.now();
+        strip.addEventListener("wheel", scrollStrip, false);
+        function scrollStrip(ev) {
+            ev.preventDefault();
+            ev.stopPropagation();
+			
+			if (Date.now() - lastScroll < 200) {
```
Comment (MxtOUT) [inline js/content/common.js]: Here

File: js/content/common.js [hunk 2]
```diff
@@ -2282,5 +2282,34 @@ class MediaPage {
...
+				return;
+			} 
+    
+            lastScroll = Date.now();
+            let allElem = document.querySelectorAll(".highlight_strip_item");
+			let isScrollDown = ev.deltaY > 0;
```
Comment (MxtOUT) [inline js/content/common.js]: Here

File: js/content/common.js [hunk 3]
```diff
@@ -2282,5 +2282,34 @@ class MediaPage {
...
+			
+            let targetElem = document.querySelector(".highlight_strip_item.focus")[siblingProp];
+            while (!targetElem.classList || !targetElem.classList.contains("highlight_strip_item")) {
+                targetElem = targetElem[siblingProp];
+                if (!targetElem) {
+                    targetElem = allElem[isScrollDown ? allElem.length - 1: 0];
+                }
+            }
+            
+			targetElem.click();
```
Comment (MxtOUT) [inline js/content/common.js]: Here

**Comment by Revadike** [comment]
Hmm, seems I somehow used tabs instead of spaces. Weird, since vscode usually always uses spaces for me.

**Comment by Revadike** [comment]
Fixed formatting

**Comment by MxtOUT** [comment]
I'd actually expect that it scrolls to the right if I scroll down, because scrolling down normally means to advance and not to go back

**Comment by Revadike** [comment]
> I'd actually expect that it scrolls to the right if I scroll down, because scrolling down normally means to advance and not to go back

That's not really the standard. Just check this:
1) Open https://wallpapercave.com/wp/wp2529606.jpg
2) Zoom in
3) Hold shift and scroll

You'll notice that scolling down will go to the left. This is standard behavior.

**Comment by MxtOUT** [comment]
https://forum.cockos.com/showthread.php?t=209679

**Comment by Ryzhehvost** [comment]
>You'll notice that scolling down will go to the left.
Well, it goes right for me.

**Comment by MxtOUT** [comment]
For me too

**Comment by Revadike** [comment]
Well, that's weird. I just reinstalled windows actually and this is all default for me. I didn't change anything.

**Comment by Revadike** [comment]
Fixed

**Pull request merged**